modules/surface_matching/src/c_utils.hpp

```diff
@@ -55,515 +55,209 @@ const float EPS = 1.192092896e-07F;        /* smallest such that 1.0+FLT_EPSILON
-  t[2] = row3[0] * (-p1[0]) + row3[1] * (-p1[1]) + row3[2] * (-p1[2]);
+  Mat(Rx * (Ry * Rz)).copyTo(R);
 }
 
+
 /**
- *  \brief Compute the transformation needed to rotate n1 onto x axis and p1 to origin
- */
-static inline void computeTransformRT(const double p1[4], const double n1[4], double R[9], double t[3])
+*  \brief Compute the transformation needed to rotate n1 onto x axis and p1 to origin
+*/
+static inline void computeTransformRT(const Vec3d& p1, const Vec3d& n1, Matx33d& R, Vec3d& t)
 {
   // dot product with x axis
-  double angle=acos( n1[0] );
+  double angle = acos(n1[0]);
 
   // cross product with x axis
-  double axis[3]={0, n1[2], -n1[1]};
-  double axisNorm;
-  double *row1, *row2, *row3;
+  Vec3d axis(0, n1[2], -n1[1]);
 
   // we try to project on the ground plane but it's already parallel
-  if (n1[1]==0 && n1[2]==0)
+  if (n1[1] == 0 && n1[2] == 0)
   {
-    axis[1]=1;
-    axis[2]=0;
+    axis[1] = 1;
+    axis[2] = 0;
   }
   else
   {
-    axisNorm=sqrt(axis[2]*axis[2]+axis[1]*axis[1]);
-
-    if (axisNorm>EPS)
-    {
-      axis[1]/=axisNorm;
-      axis[2]/=axisNorm;
-    }
+    TNormalize3(axis);
   }
 
-  aaToR(angle, axis, R);
-  row1 = &R[0];
-  row2 = &R[3];
-  row3 = &R[6];
-
-  t[0] = row1[0] * (-p1[0]) + row1[1] * (-p1[1]) + row1[2] * (-p1[2]);
-  t[1] = row2[0] * (-p1[0]) + row2[1] * (-p1[1]) + row2[2] * (-p1[2]);
-  t[2] = row3[0] * (-p1[0]) + row3[1] * (-p1[1]) + row3[2] * (-p1[2]);
+  aaToR(axis, angle, R);
+  t = -R * p1;
 }
 
 /**
  *  \brief Flip a normal to the viewing direction
  *
  *  \param [in] point Scene point
- *  \param [in] vp_x X component of view direction
- *  \param [in] vp_y Y component of view direction
- *  \param [in] vp_z Z component of view direction
- *  \param [in] nx X component of normal
- *  \param [in] ny Y component of normal
- *  \param [in] nz Z component of normal
- */
-static inline void flipNormalViewpoint(const float* point, double vp_x, double vp_y, double vp_z, double *nx, double *ny, double *nz)
-{
-  double cos_theta;
-
-  // See if we need to flip any plane normals
-  vp_x -= (double)point[0];
-  vp_y -= (double)point[1];
-  vp_z -= (double)point[2];
-
-  // Dot product between the (viewpoint - point) and the plane normal
-  cos_theta = (vp_x * (*nx) + vp_y * (*ny) + vp_z * (*nz));
-
-  // Flip the plane normal
-  if (cos_theta < 0)
-  {
-    (*nx) *= -1;
-    (*ny) *= -1;
-    (*nz) *= -1;
-  }
-}
-/**
- *  \brief Flip a normal to the viewing direction
- *
- *  \param [in] point Scene point
- *  \param [in] vp_x X component of view direction
- *  \param [in] vp_y Y component of view direction
- *  \param [in] vp_z Z component of view direction
- *  \param [in] nx X component of normal
- *  \param [in] ny Y component of normal
- *  \param [in] nz Z component of normal
+ *  \param [in] vp view direction
+ *  \param [in] n normal
  */
-static inline void flipNormalViewpoint_32f(const float* point, float vp_x, float vp_y, float vp_z, float *nx, float *ny, float *nz)
```

Comment:
There is no need for the float version, since the double version can handle things